### PR TITLE
add text validation

### PR DESCRIPTION
**Pull request by haysho2260**
- took about 2.5 hours
- i took longer because had some cached data that was messing up page
- validated text data
- added validations message
- made sure number had minimum

### CODE REVIEW

File: react-frontend/src/components/LogWorkout.js [hunk 1]
```diff
@@ -105,9 +134,17 @@ function LogWorkout({ handleLogWorkoutSubmit, ...props }) {
                     <Offcanvas.Body style={{ paddingTop: "8px" }}>
                          <Form
                               noValidate
-                              // validated={validated}
+                              validated={validated}
                               onSubmit={handleSubmit}
                          >
+                              {/* Alert for displaying validation errors */}
```
Comment (joanna-chou) [inline react-frontend/src/components/LogWorkout.js:140]: I like the alert but this is the only place where an alert is implemented. If we want to be consistent / less work as to not change all the other pages maybe we can just keep the input validation ui and not use an alert.

File: react-frontend/src/components/WorkoutCard.js [hunk 2]
```diff
@@ -6,6 +6,12 @@ import "./WorkoutCard.scss";
 
 // Create content in card
 function CardTable(props) {
+     console.log(props.item);
+     // Check if props.item is an array
```
Comment (joanna-chou) [inline react-frontend/src/components/WorkoutCard.js:10]: We resolved this issue right. I think there will be other failsafes in place after we move from local storage to databases.

**Review by jqueja — APPROVED** [review]
Great Use of Form Submission logic to alert the user.
Great use of commends and documentation.

Overall great job!

**Comment by haysho2260** [comment]
@Parshana007 for your comment should we make a new issue about fixing backend to the new edits?

File: react-frontend/src/components/LogWorkout.js (hunk 1 above)
Comment (haysho2260) [reply react-frontend/src/components/LogWorkout.js:121]: so should i delete it?

Comment (haysho2260) [reply react-frontend/src/components/LogWorkout.js:121]: Follow up: asked in person and mentioned that it was ok to keep

File: react-frontend/src/components/WorkoutCard.js (hunk 2 above)
Comment (haysho2260) [reply react-frontend/src/components/WorkoutCard.js]: yeah but i don't think it hurts to have it right?

**Pull request merged**